ooniapi/services/oonimeasurements/src/oonimeasurements/routers/data/utils.py

```diff
@@ -50,3 +52,11 @@ def test_name_to_group(tn):
         return "websites"
     # TODO(arturo): currently we only support websites
     return ""
+
+
+def parse_probe_asn(probe_asn):
```

Comment:
Maybe it's a good idea to add typing to this argument to clarify that this function is correct only in the case that it's taking as argument `Union[str, int, None]` which is validated by the fastapi type annotation spec.

Reply:
Yes, but this is related to the following comment, at the very least it should be typed as a string. I will modify.

Reply:
Can't it be an `int` though? In the previous version of this we were checking if it's `str` AND `startsWith(AS)` and then doing the casting to `int`: https://github.com/ooni/backend/pull/921/files/9d0c56f0f649a4b48bfa7b330d8f5c02a3b4d265#diff-e06ff97cf1030d17023c1d12d21537165829bb6cf53336da5cec4f95d43af526L100.

The fact it can be `Union[str, int, None]` comes from the types specified in the fastapi spec: https://github.com/ooni/backend/pull/921/files/9d0c56f0f649a4b48bfa7b330d8f5c02a3b4d265#diff-facb5f0fb26ad641ef437fcad578b9c36afe385edcf5cb27a3f590449c2ef856R171

Reply:
Yes, I think you are right! It should work with int too. But I think I found a bug with list_analysis regarding this. I'm going to create tests filtering with integers to verify that they should work correctly. I'll let you know when there are any changes.